src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -444,6 +491,20 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
     }
 
     /**
+     * Checks if right curly is alone on line in token the lambda.
+     * @param details for validation.
+     * @return true, if right curly is alone on line.
+     */
+    private static boolean isAloneInLambda(Details details) {
+        final DetailAST lcurly = details.lcurly;
+        final DetailAST rcurly = details.rcurly;
+        final int lcurlyNo = lcurly.getLineNo();
+        final int rcurlyNo = rcurly.getLineNo();
+        return rcurlyNo != lcurlyNo
+            && rcurlyNo != rcurly.getPreviousSibling().getLineNo();
```

Comment:
This is shown as outdated, but the way you changed it is now leading to a TC failure.
`504: isAloneInLambda() if statement with identical branches`